processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -45,203 +67,911 @@ impl AuxTraceBuilder {
 // ================================================================================================
 
 /// Describes how to construct the execution trace of the chiplets bus auxiliary trace column.
-pub struct BusTraceBuilder {
-    pub(super) lookup_hints: Vec<(u32, ChipletsBusRow)>,
-    pub(super) requests: Vec<ChipletLookup>,
-    pub(super) responses: Vec<ChipletLookup>,
-}
-
-impl BusTraceBuilder {
-    pub(crate) fn new(
-        lookup_hints: Vec<(u32, ChipletsBusRow)>,
-        requests: Vec<ChipletLookup>,
-        responses: Vec<ChipletLookup>,
-    ) -> Self {
-        Self {
-            lookup_hints,
-            requests,
-            responses,
-        }
+#[derive(Default)]
+pub struct BusTraceBuilder {}
+
+/// Builds the chiplets bus auxiliary trace column.
+fn build_bus_aux_column<E>(main_trace: &ColMatrix<Felt>, alphas: &[E]) -> Vec<E>
+where
+    E: FieldElement<BaseField = Felt>,
+{
+    let main_tr = MainTrace::new(main_trace);
+    let mut result_1: Vec<E> = unsafe { uninit_vector(main_trace.num_rows()) };
+    let mut result_2: Vec<E> = unsafe { uninit_vector(main_trace.num_rows()) };
+    result_1[0] = E::ONE;
+    result_2[0] = E::ONE;
+
+    let mut result_2_acc = E::ONE;
+    for i in 0..main_trace.num_rows() - 1 {
```

Comment:
Currently, every chiplet performs its own loop over the entire trace. It would probably be more cache-efficient to have just one loop, in which we every chiplet performs its work in the body - but we'd need to measure the performance delta to be sure.

It would also make further optimizations easier, such as parallelizing the loop (as opposed to parallelize `n` loops for `n` chiplets).